src/tree/measure_func.rs

```diff
@@ -8,8 +8,18 @@ use crate::util::sys::Box;
 /// A function type that can be used in a [`MeasureFunc`]
 ///
 /// This trait is automatically implemented for all types (including closures) that define a function with the appropriate type signature.
-pub trait Measurable: Send + Sync + Fn(Size<Option<f32>>, Size<AvailableSpace>) -> Size<f32> {}
-impl<F: Send + Sync + Fn(Size<Option<f32>>, Size<AvailableSpace>) -> Size<f32>> Measurable for F {}
+pub trait Measurable {
+    /// Measure node
+    fn measure(&self, known_dimensions: Size<Option<f32>>, available_space: Size<AvailableSpace>) -> Size<f32>;
+}
+/// Version of the Measurable trait that also requires Send + Sync
+pub trait SyncMeasurable: Send + Sync + Measurable {}
+impl<F: Fn(Size<Option<f32>>, Size<AvailableSpace>) -> Size<f32>> Measurable for F {
+    fn measure(&self, known_dimensions: Size<Option<f32>>, available_space: Size<AvailableSpace>) -> Size<f32> {
+        self(known_dimensions, available_space)
+    }
+}
+impl<F: Send + Sync + Fn(Size<Option<f32>>, Size<AvailableSpace>) -> Size<f32>> SyncMeasurable for F {}
```

Comment:
For the time being could you:
- Add the Send + Sync bounds back to the `Measurable` trait
- Delete the `SyncMeasurable` trait

I do want to introduce this distinction, but I think it should be a separate change.